lisa/executable.py

```diff
@@ -0,0 +1,383 @@
+from __future__ import annotations
+
+import pathlib
+import re
+from abc import ABC, abstractmethod
+from hashlib import sha256
+from typing import TYPE_CHECKING, List, Optional, Type, Union, cast
+
+from lisa.util.exceptions import LisaException
+from lisa.util.process import ExecutableResult, Process
+
+if TYPE_CHECKING:
+    from lisa.node import Node
+
+
+class Tool(ABC):
+    """
+    The base class, which wraps an executable, package, or scripts on a node.
+    A tool can be installed, and execute on a node. When a tool is needed, call
+    Node.getTool() to get one object. The getTool checks if it's installed. If it's
```

Comment:
This is different with envrionment. The getTool accepts generic type as parameter, and returns an instance same type. So that user can the prompt methods when developing. If [] can do this also, it can be changed. If so, please show me an example.

Reply:
I have a try, and __getitem__ supports generic type also. updated.